### PR TITLE
add fakecast and fix split key issues

### PR DESCRIPTION
**Pull request by evanmcc**
There are two parts of this PR.  The three commits up to c4e52a8, and the fix in commit ea7de92.  See https://app.clubhouse.io/hlm/story/910/qbar-concerns for more history and details.

1. I wrote fakecast to be able to get at some weirder cases.  The VSS "test" is really just a replication of the tests that use relcast.  It isn't very interesting, and could potentially be deleted.  Mostly I wanted to be able to make sure that it could run a relcast protocol correctly.

c4e52a8 adds the test that manages to make the dkg produce split keys.

2.  After reading the traces and adding some debugging input, I noticed that the shares proposal grew quite a lot, when it should likely never be bigger than `T+1` shares.  Working from there, I noted some infelicities with the protocol as outlined in the paper and remedied them.

### CODE REVIEW

**Comment by evanmcc** [comment]
[ch910]

File: test/dkg_hybridvss_SUITE.erl [hunk 1]
```diff
@@ -44,6 +48,27 @@ asymmetric_test(Config) ->
...
 
+%% placeholder name
+fake_symmetric(Config) ->
+    N = proplists:get_value(n, Config),
+    F = proplists:get_value(f, Config),
+    T = proplists:get_value(t, Config),
+    Module = proplists:get_value(module, Config),
+    Curve = 'SS512',
+    {G1, G2} = dkg_test_utils:generate(Curve),
+    ok = run_fake(Module, N, F, T, Curve, G1, G2),
```
Comment (Vagabond) [inline test/dkg_hybridvss_SUITE.erl]: Why is this called twice in both of these tests?

File: test/dkg_hybriddkg_SUITE.erl [hunk 2]
```diff
@@ -67,6 +69,15 @@ leader_change_asymmetric_test(Config) ->
     {G1, G2} = dkg_test_utils:generate('MNT224'),
     run(N, F, T, Module, 'MNT224', G1, G2, InitialLeader).
 
+split_key_test(Config) ->
```
[the remark this answers is not in the document]
Comment (Vagabond) [reply test/dkg_hybriddkg_SUITE.erl]: Oh, I see, we change the arguments to the fakecast. Maybe add some comments here as to what you're doing?

File: test/dkg_hybridvss_SUITE.erl (hunk 1 above)
Comment (evanmcc) [reply test/dkg_hybridvss_SUITE.erl]: oops!  I was seeing if it doubled, timing-wise, and forgot to take it out.

File: src/dkg_hybriddkg.erl [hunk 3]
```diff
@@ -112,13 +115,42 @@ init(Id, N, F, T, G1, G2, Round, Options) ->
...
+    start(State).
+
 start(DKG = #dkg{id=Id, u=G1}) ->
     #{Id := MyShares} = SharesMap = DKG#dkg.shares_map,
     Secret = erlang_pbc:element_random(erlang_pbc:element_new('Zr', G1)),
     {NewShares, {send, ToSend}} = dkg_hybridvss:input(MyShares, Secret),
     {DKG#dkg{shares_map = SharesMap#{Id => NewShares}},
      {send, dkg_util:wrap({vss, Id}, ToSend)}}.
 
+handle_msg(DKG=#dkg{await_vss = true}, Sender, {{vss, SharesId}, SharesMsg}) ->
```
Comment (vihu) [inline src/dkg_hybriddkg.erl]: I noticed while running the tests locally that this clause is pretty much uncovered?

Comment (evanmcc) [reply src/dkg_hybriddkg.erl]: I'll do my best to extend coverage here later, but at the moment I'm not sure it's worth the effort for such a rare case.

**Comment by evanmcc** [comment]
updated for fakecast v2

File: src/dkg_hybriddkg.erl (hunk 3 above)
Comment (vihu) [reply src/dkg_hybriddkg.erl]: Fair enough, I know triggering this particular branch is probably a lot trickier as we discussed offline

**Pull request merged**